task_sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -479,12 +481,39 @@ def run(ti: RuntimeTaskInstance, log: Logger):
 
         _push_xcom_if_needed(result, ti)
 
+        task_outlets = []
+        outlet_events = []
+        events = context["outlet_events"]
+
+        for obj in ti.task.outlets or []:
+            # Lineage can have other types of objects besides assets
+            asset_type = type(obj).__name__
+            if isinstance(obj, Asset):
+                task_outlets.append(AssetProfile(name=obj.name, uri=obj.uri, asset_type=asset_type))
+                outlet_events.append(attrs.asdict(events[obj]))  # type: ignore
+            elif isinstance(obj, AssetNameRef):
+                task_outlets.append(AssetProfile(name=obj.name, asset_type=asset_type))
+                # Send all events, filtering can be done in API server.
+                outlet_events.append(attrs.asdict(events))  # type: ignore
+            elif isinstance(obj, AssetUriRef):
+                task_outlets.append(AssetProfile(uri=obj.uri, asset_type=asset_type))
+                # Send all events, filtering can be done in API server.
+                outlet_events.append(attrs.asdict(events))  # type: ignore
+            elif isinstance(obj, AssetAlias):
+                task_outlets.append(AssetProfile(asset_type=asset_type))
+                for asset_alias_event in events[obj].asset_alias_events:
+                    outlet_events.append(attrs.asdict(asset_alias_event))
```

Comment:
Each of these branches have asset types defined even if name and uri are optional.

Reply:
For clarity lets pull this all out into a func called something like `process_assets` or `process_outlets`

Reply:
Yeah let me do that. I have similar logic in task runner too. 

Wondering if I can move it to lets say `definitions/assets/utils.py`? Definitions doesn't look to be a good place to me honestly, but I cant think of some other location. Any suggestions?

Reply:
It can be in this file I think -- it just makes it easier to understand what it does if it's got a name 😄

Reply:
Yeah, kept it there itself.